unlock-app/app/frames/checkout/approve/route.ts

```diff
@@ -0,0 +1,29 @@
+import { Abi, encodeFunctionData, erc20Abi } from 'viem'
+import { frames } from '../frames'
+import { transaction } from 'frames.js/core'
+
+export const POST = frames(async (ctx) => {
+  if (!ctx?.message) {
+    throw new Error('Invalid frame message')
+  }
+
+  const lock = ctx.state.lock!
+  const { address: lockAddress, network, priceForUser } = lock
+
+  const calldata = encodeFunctionData({
```

Comment:
You don't need to use that! unlock-js has the logic you need to use.

Reply:
So I think you should first check if the user has already approved enough (and you can compute the amount that needs to be approved based on the ehckout config (especially if it supports renewals)

Sorry my initial comment was unclear. Check unlock-js where there is already logic around how to compute the amount to be approved.